packages/turbo/ios/RNSession.swift

```diff
@@ -61,102 +61,86 @@ class RNSession: NSObject {
   }()
   public lazy var webView: WKWebView = turboSession.webView
   
-  func registerVisitableView(newView: RNSessionSubscriber) {
-    if (!visitableViews.contains {
-      $0.id == newView.id
-    }) {
-      visitableViews.append(newView)
+  func visitableViewDidDisappear(view: RNVisitableView) {
+    if (view === visitableView) {
+      self.visitableView = nil
     }
   }
   
-  func unregisterVisitableView(view: RNSessionSubscriber) {
-    let wasTopMostView = visitableViews.last?.id == view.id
-    
-    let viewIdx = visitableViews.lastIndex(where: {
-      view.id == $0.id
-    })
-    visitableViews.remove(at: viewIdx!)
-
-    // The new top-most view is not registered when the previous top-most view is a modal
-    if (wasTopMostView) {
-      guard let newView = visitableViews.last else {
-        return
-      }
-      visitableViewWillAppear(view: newView)
-    }
-  }
+  func visitableViewDidAppear(view: RNVisitableView) {
+    // if (visitableView !== nil && visitableView.isModal !== view.isModal) {
+    //   print("You're not able to share session between modal and non-modals.")
+    // }
```

Comment:
Not sure if we should address this in the PR?

Reply:
We can cover that in this PR since it is breaking previous behaviour but I would prefer to state that in a docs rather than throwing/printing an error on the native side.